openquake/hazardlib/gsim/gsim_table.py

```diff
@@ -525,7 +525,7 @@ def _return_tables(self, mag, imt, val_type):
             low_period = round(periods[0], 7)
             high_period = round(periods[-1], 7)
 
-            if imt.period < low_period or imt.period > high_period:
+            if round(imt.period, 7) < low_period or round(imt.period, 7) > high_period:
```

Comment:
Probably this is not the only point at which precision issues may cause problems. The fix looks good to me, and I guess we will find the others in due course.